src/Orchard.Web/Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -60,55 +57,54 @@ public class AutoroutePartHandler : ContentHandler {
-            if (part.DisplayAlias == "/") {
-                part.DisplayAlias = String.Empty;
-
-                // regenerate the alias for the previous home page
-                var currentHomePages = _orchardServices.ContentManager.Query<AutoroutePart, AutoroutePartRecord>().Where(x => x.DisplayAlias == "").List();
-                foreach (var current in currentHomePages.Where(x => x.Id != part.Id)) {
-                    if (current != null) {
-                        current.CustomPattern = String.Empty; // force the regeneration
-                        current.DisplayAlias = _autorouteService.Value.GenerateAlias(current);
-
-                        // we changed the alias of the previous homepage, so publish this change if the content item was published.
-                        if(current.IsPublished())
-                            _orchardServices.ContentManager.Publish(current.ContentItem);
+            // Should it become the home page?
+            if (part.PromoteToHomePage) {
+                // Get the current homepage an unmark it as the homepage.
+                var currentHomePage = _homeAliasService.GetHomePage(VersionOptions.Latest);
+                if(currentHomePage != null && currentHomePage.Id != part.Id) {
+                    var autoroutePart = currentHomePage.As<AutoroutePart>();
+
+                    if (autoroutePart != null) {
+                        autoroutePart.PromoteToHomePage = false;
+                        if(autoroutePart.IsPublished())
+                            _orchardServices.ContentManager.Publish(autoroutePart.ContentItem);
                     }
-                    _autorouteService.Value.PublishAlias(current);
                 }
+
+                // Update the home alias to point to this item being published.
+                _homeAliasService.PublishHomeAlias(part);
             }
 
             _autorouteService.Value.PublishAlias(part);
         }
 
         private void ProcessAlias(AutoroutePart part) {
-            // generate an alias if one as not already been entered
+            // Generate an alias if one as not already been entered.
             if (String.IsNullOrWhiteSpace(part.DisplayAlias)) {
                 part.DisplayAlias = _autorouteService.Value.GenerateAlias(part);
             }
 
-            // if the generated alias is empty, compute a new one 
+            // If the generated alias is empty, compute a new one.
             if (String.IsNullOrWhiteSpace(part.DisplayAlias)) {
                 _autorouteService.Value.ProcessPath(part);
                 _orchardServices.Notifier.Warning(T("The permalink could not be generated, a new slug has been defined: \"{0}\"", part.Path));
                 return;
             }
 
-            // check for permalink conflict, unless we are trying to set the home page
+            // Check for permalink conflict, unless we are trying to set the home page.
             if (part.DisplayAlias != "/") {

```

Comment:
Actually, we should not even check for the content being the homepage, since we should always check if the alias is in conflict with an existing one. A content item no longer is the home page if its display alias is "", but simply by virtue of the "/" alias record pointing to the content item. Another good catch.